infoschema/perfschema/tables.go

```diff
@@ -77,3 +86,135 @@ func (vt *perfSchemaTable) GetPhysicalID() int64 {
 func (vt *perfSchemaTable) Meta() *model.TableInfo {
 	return vt.meta
 }
+
+func (vt *perfSchemaTable) getRows(ctx sessionctx.Context, cols []*table.Column) (fullRows [][]types.Datum, err error) {
+	switch vt.meta.Name.O {
+	case tableNameEventsStatementsSummaryByDigest:
+		fullRows = dataForEventsStatementsSummaryByDigest(ctx)
+	}
+	if len(cols) == len(vt.cols) {
+		return
+	}
+	rows := make([][]types.Datum, len(fullRows))
+	for i, fullRow := range fullRows {
+		row := make([]types.Datum, len(cols))
+		for j, col := range cols {
+			row[j] = fullRow[col.Offset]
+		}
+		rows[i] = row
+	}
+	return rows, nil
+}
+
+// IterRecords implements table.Table IterRecords interface.
+func (vt *perfSchemaTable) IterRecords(ctx sessionctx.Context, startKey kv.Key, cols []*table.Column,
+	fn table.RecordIterFunc) error {
+	if len(startKey) != 0 {
+		return table.ErrUnsupportedOp
+	}
+	rows, err := vt.getRows(ctx, cols)
+	if err != nil {
+		return err
+	}
+	for i, row := range rows {
+		more, err := fn(int64(i), row, cols)
+		if err != nil {
+			return err
+		}
+		if !more {
+			break
+		}
+	}
+	return nil
+}
+
+// RowWithCols implements table.Table RowWithCols interface.
+func (vt *perfSchemaTable) RowWithCols(ctx sessionctx.Context, h int64, cols []*table.Column) ([]types.Datum, error) {
+	return nil, table.ErrUnsupportedOp
+}
+
+// Row implements table.Table Row interface.
+func (vt *perfSchemaTable) Row(ctx sessionctx.Context, h int64) ([]types.Datum, error) {
+	return nil, table.ErrUnsupportedOp
+}
+
+// Indices implements table.Table Indices interface.
+func (vt *perfSchemaTable) Indices() []table.Index {
+	return nil
+}
+
+// WritableIndices implements table.Table WritableIndices interface.
+func (vt *perfSchemaTable) WritableIndices() []table.Index {
+	return nil
+}
+
+// DeletableIndices implements table.Table DeletableIndices interface.
+func (vt *perfSchemaTable) DeletableIndices() []table.Index {
+	return nil
+}
+
+// RecordPrefix implements table.Table RecordPrefix interface.
+func (vt *perfSchemaTable) RecordPrefix() kv.Key {
+	return nil
+}
+
+// IndexPrefix implements table.Table IndexPrefix interface.
+func (vt *perfSchemaTable) IndexPrefix() kv.Key {
+	return nil
+}
+
+// FirstKey implements table.Table FirstKey interface.
+func (vt *perfSchemaTable) FirstKey() kv.Key {
+	return nil
+}
+
+// RecordKey implements table.Table RecordKey interface.
+func (vt *perfSchemaTable) RecordKey(h int64) kv.Key {
+	return nil
+}
+
+// AddRecord implements table.Table AddRecord interface.
+func (vt *perfSchemaTable) AddRecord(ctx sessionctx.Context, r []types.Datum, opts ...table.AddRecordOption) (recordID int64, err error) {
+	fmt.Println("add record")
```

Comment:
Please remove this manual-debugging code

Reply:
Addressed.